test/unittests/libp2p/net.cpp

```diff
@@ -52,292 +52,387 @@ BOOST_FIXTURE_TEST_SUITE(net, NetFixture)
+                return bucketIndex;
+
+            ++testNode;
+        }
+
+        // not enough testNodes
+        return -1;
+    }
+
+    void reset()
+    {
+        Guard l(x_state);
+        for (auto& n: m_state) n.nodes.clear();
+    }
+
+    using NodeTable::m_evictions;
+    using NodeTable::m_nodes;
+    using NodeTable::m_state;
+    using NodeTable::noteActiveNode;
 };
 
 /**
  * Only used for testing. Not useful beyond tests.
  */
 struct TestNodeTableHost: public TestHost
 {
-	TestNodeTableHost(unsigned _count = 8): m_alias(KeyPair::create()), nodeTable(new TestNodeTable(m_io, m_alias, bi::address::from_string("127.0.0.1"))), testNodes(TestNodeTable::createTestNodes(_count)) {};
-	~TestNodeTableHost() { m_io.stop(); stopWorking(); }
+    TestNodeTableHost(unsigned _count = 8): m_alias(KeyPair::create()), nodeTable(new TestNodeTable(m_io, m_alias, bi::address::from_string("127.0.0.1"))), testNodes(TestNodeTable::createTestNodes(_count)) {};
+    ~TestNodeTableHost() { m_io.stop(); stopWorking(); }
 
-	void setup() { for (auto n: testNodes) nodeTables.push_back(make_shared<TestNodeTable>(m_io,n.first, bi::address::from_string("127.0.0.1"),n.second)); }
+    void populate(size_t _count = 0) { nodeTable->populateTestNodes(testNodes, _count); }
 
-	void pingAll() { for (auto& t: nodeTables) t->pingTestNodes(testNodes); }
+    int populateUntilBucketSize(size_t _size)
+    {
+        return nodeTable->populateUntilBucketSize(testNodes, _size);
+    }
 
-	void populateAll(size_t _count = 0) { for (auto& t: nodeTables) t->populateTestNodes(testNodes, _count); }
-
-	void populate(size_t _count = 0) { nodeTable->populateTestNodes(testNodes, _count); }
-
-	KeyPair m_alias;
-	shared_ptr<TestNodeTable> nodeTable;
-	std::vector<std::pair<KeyPair,unsigned>> testNodes; // keypair and port
-	std::vector<shared_ptr<TestNodeTable>> nodeTables;
+    KeyPair m_alias;
+    shared_ptr<TestNodeTable> nodeTable;
+    std::vector<std::pair<Public, unsigned>> testNodes;  // keypair and port
 };
 
-class TestUDPSocket: UDPSocketEvents, public TestHost
+class TestUDPSocketHost : UDPSocketEvents, public TestHost
 {
 public:
-	TestUDPSocket(unsigned _port): m_socket(new UDPSocket<TestUDPSocket, 1024>(m_io, *this, _port)) {}
+    TestUDPSocketHost(unsigned _port)
+      : m_socket(new UDPSocket<TestUDPSocketHost, 1024>(m_io, *this, _port))
+    {}
 
-	void onDisconnected(UDPSocketFace*) {};
-	void onReceived(UDPSocketFace*, bi::udp::endpoint const&, bytesConstRef _packet) { if (_packet.toString() == "AAAA") success = true; }
+    void onDisconnected(UDPSocketFace*) {};
+    void onReceived(UDPSocketFace*, bi::udp::endpoint const&, bytesConstRef _packet) { if (_packet.toString() == "AAAA") success = true; }
 
-	shared_ptr<UDPSocket<TestUDPSocket, 1024>> m_socket;
+    shared_ptr<UDPSocket<TestUDPSocketHost, 1024>> m_socket;
 
-	bool success = false;
+    bool success = false;
 };
 
-BOOST_AUTO_TEST_CASE(requestTimeout)
```

Comment:
Removing because it doesn't look like testing any of our code, just checking how `std::list::remove_if` works